ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -146,7 +147,7 @@ public void run() {
 
               // Compaction doesn't work under a transaction and hence pass null for validTxnList
               // The response will have one entry per table and hence we get only one ValidWriteIdList
-              String fullTableName = TxnUtils.getFullTableName(t.getDbName(), t.getTableName());
+              String fullTableName = Warehouse.getQualifiedName(t.getDbName(), t.getTableName());
```

Comment:
Shouldn't this include catalog in the name?  Looks like it will end up in TxnHandler.getValidWriteIdsForTable() which does FullTableName tableName = new FullTableName(fullTableName, conf); which assume a default catalog name.
Couldn't it be in some other cat?

Reply:
Agreed.